core/src/main/java/org/apache/druid/java/util/common/ByteBufferUtils.java

```diff
@@ -138,6 +144,45 @@ private static MethodHandle unmapJava9(MethodHandles.Lookup lookup) throws Refle
     return unmapper.bindTo(UnsafeUtils.theUnsafe());
   }
 
+  /**
+   * Same as {@link ByteBuffer#allocateDirect(int)}, but returns a closeable {@link ResourceHolder} that
+   * frees the buffer upon close.
+   */
+  public static ResourceHolder<ByteBuffer> allocateDirect(final int size)
```

Comment:
Should we also use this utility method in places such as `LZ4Compressor.allocateInBuffer()` rather than calling `free` directly from there?